src/overlays/actors/ovl_Boss_Fd/z_boss_fd.c

```diff
@@ -21,51 +105,1972 @@ const ActorInit Boss_Fd_InitVars = {
+
+                this->actionState = 0;
+                this->animationTimers[0] = 0;
+                this->startAttack = 0;
+            } else {
+                Math_SmoothScaleMaxF(&this->targetLinkYOffset, 50.0, 1.0f, 2.0f);
+            }
+            break;
+        case 200: // Start of death cutscene
+            if (sqrtf(dx * dx + dz * dz) < 50.0f) {
+                this->animationTimers[0] = 0;
+            }
+            if (this->animationTimers[0] == 0) {
+                this->animationTimers[0] = (s16)Math_Rand_ZeroFloat(10.0f) + 10;
+                do {
+                    this->targetPosition.x = Math_Rand_CenteredFloat(200.0f);
+                    this->targetPosition.y = 390.0f;
+                    this->targetPosition.z = Math_Rand_CenteredFloat(200.0f);
+                    temp_x = this->targetPosition.x - this->actor.posRot.pos.x;
+                    temp_z = this->targetPosition.z - this->actor.posRot.pos.z;
+                } while (!(sqrtf((temp_x * temp_x) + (temp_z * temp_z)) > 100.0f));
+            }
+            this->flightWobbleAmplitude = 200.0f;
+            this->flightWobbleRate = 1000.0f;
+            this->maxTurnRate = 10000.0f;
+            Math_SmoothScaleMaxF(&this->bodyPulse, 0.3f, 1.0f, 0.05f);
+            if (this->animationTimers[1] == 0) {
+                this->actionState = 201;
+                this->animationTimers[0] = 30;
+            }
+            break;
+        case 201: // Skin burning off scene
+            this->targetPosition.x = 0.0f;
+            this->targetPosition.y = 390.0f;
+            this->targetPosition.z = 0.0f;
+            this->flightWobbleAmplitude = 200.0f;
+            this->flightWobbleRate = 1000.0f;
+            this->maxTurnRate = 2000.0f;
+            Math_SmoothScaleMaxF(&this->bodyPulse, 0.3f, 1.0f, 0.05f);
+            if ((this->animationTimers[0] == 0) && ((this->movementTimer & 3) == 0)) {
+                if (this->skinSegments != 0) {
+                    this->skinSegments--;
+                    if (this->skinSegments == 0) {
+                        Audio_SetBGM(0x21);
+                    }
+                } else {
+                    this->actionState = 202;
+                    this->animationTimers[0] = 30;
+                }
+            }
+            if ((this->movementTimer & 0x1F) == 0) {
+                this->roarTimer = 40;
+            }
+
+            if (this->skinSegments != 0) {
+                sp17C = D_808D19E0;
+                sp164 = D_808D19EC;
+
+                if (this->fogMode == 0) {
+                    globalCtx->envCtx.unk_D8 = 0;
+                }
+                this->fogMode = 0xA;
+
+                sp150 = 1;
+                if (this->movementTimer & 0x1C) {
+                    Audio_PlaySoundGeneral(0x30DA, &this->actor.projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                }
+                for (i1 = 0; i1 < sp150; i1++) {
+                    if (sp150) { // needed for match
```

Comment:
can this `if` not wrap everything in the for, like, is `if (sp150) {}` enough?

Reply:
It is not. This took ages to find a configuration that matched.